src/main/java/be/seeseemelk/mockbukkit/RegistryMock.java

```diff
@@ -0,0 +1,81 @@
+package be.seeseemelk.mockbukkit;
+
+import io.papermc.paper.world.structure.ConfiguredStructure;
+import org.bukkit.Keyed;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
+import org.bukkit.generator.structure.Structure;
+import org.bukkit.generator.structure.StructureType;
+import org.bukkit.inventory.meta.trim.TrimMaterial;
+import org.bukkit.inventory.meta.trim.TrimPattern;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.ParameterizedType;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+public class RegistryMock
+{
+
+	public static <T extends Keyed> Registry<?> createRegistry(Class<T> tClass)
+	{
+
+		if (tClass == Structure.class
+				|| tClass == StructureType.class
+				|| tClass == TrimMaterial.class
+				|| tClass == TrimPattern.class
+				|| tClass == ConfiguredStructure.class)
+		{
+			return new Registry<>()
+			{
+				@Override
+				public @Nullable Keyed get(@NotNull NamespacedKey key)
+				{
+					throw new UnimplementedOperationException("Registry for type " + tClass + " not implemented");
+				}
+
+				@NotNull
+				@Override
+				public Iterator<Keyed> iterator()
+				{
+					throw new UnimplementedOperationException("Registry for type " + tClass + " not implemented");
+				}
+			};
+		}
```

Comment:
This isn't necessary, as an UnimplementedOperationException will be thrown at the end of the following stream. It's better to throw the exception when trying to create the registry rather than stubbing it.

Reply:
It is necessary.

Within the class Registry theres a circular reference

```
Registry<Structure> STRUCTURE = Bukkit.getRegistry(Structure.class);
```
Registry class initializes -> Bukkit.getRegistry -> servermock.getregistry -> get field value -> ... 

Theres no other way around it, unless you change code in bukkit api directly.

Reply:
I see. Can we still throw the exception here instead of returning a stubbed class?

Reply:
Oh, I see. I misunderstood your first explanation.